src/ragger/conftest/base_conftest.py

```diff
@@ -91,21 +91,14 @@ def pytest_generate_tests(metafunc):
         device = metafunc.config.getoption("device")
         backend_name = metafunc.config.getoption("backend")
 
-        if device == "all":
-            if backend_name != "speculos":
-                raise ValueError("Invalid device parameter on this backend")
-
-            # Add all supported firmwares
-            for fw in FIRMWARES:
+        # Enable firmware for requested devices
+        for fw in FIRMWARES:
+            if (device == fw.name) or (device == "all") or (device == "all_nano" and fw.name.startswith("nano")):
```

Comment:
I think you could use https://github.com/LedgerHQ/ragger/blob/master/src/ragger/firmware/structs.py#L41 ?